tests/sync/app.cpp

```diff
@@ -824,6 +827,350 @@ TEST_CASE("app: link_user integration", "[sync][app]") {
     }
 }
 
+TEST_CASE("app: remote mongo client", "[sync][app]") {
+    
+    std::unique_ptr<GenericNetworkTransport> (*factory)() = []{
+        return std::unique_ptr<GenericNetworkTransport>(new IntTestTransport);
+    };
+    std::string base_url = get_base_url();
+    std::string config_path = get_config_path();
+    REQUIRE(!base_url.empty());
+    REQUIRE(!config_path.empty());
+    auto config = App::Config{get_runtime_app_id(config_path), factory, base_url};
+    auto app = App(config);
+    std::string base_path = tmp_dir() + "/" + config.app_id;
+    reset_test_directory(base_path);
+    TestSyncManager init_sync_manager(base_path);
+    
+    auto remote_client = app.remote_mongo_client("BackingDB");
+    auto db = remote_client.db("test_data");
+    auto collection = db["Dog"];
+    
+    auto dog_document = "{\"name\":\"fido\", \"breed\":\"king charles\"}";
+    auto dog_document2 = "{\"name\":\"biscuit\", \"breed\":\"husky\"}";
+    auto random_name_document = util::format("{\"name\":\"%1\", \"breed\":\"husky\"}", random_string(10));
+
+    auto email = util::format("realm_tests_do_autoverify%1@%2.com", random_string(10), random_string(10));
+    auto password = random_string(10);
+    
+    app.provider_client<App::UsernamePasswordProviderClient>()
+    .register_email(email,
+                    password,
+                    [&](Optional<app::AppError> error) {
+                        CHECK(!error);
+                    });
+    
+    app.log_in_with_credentials(realm::app::AppCredentials::username_password(email, password),
+                                [&](std::shared_ptr<realm::SyncUser> user, Optional<app::AppError> error) {
+                                    REQUIRE(user);
+                                    CHECK(!error);
+                                });
+    
+    SECTION("insert") {
+        
+        bool processed = false;
+        std::string dog_object_id;
+        
+        collection.insert_one(dog_document,
+                              [&](std::string document_json, Optional<app::AppError> error) {
+            if (error) {
+                std::cout << "insert failed, message: " << error->error_code.message() << "+" << error->message << std::endl;
+            } else {
+                try {
+                    auto json = nlohmann::json::parse(document_json);
+                    auto object_id = json.at("insertedId").at("$oid").get<std::string>();
+                    
+                    dog_object_id = object_id;
+                } catch (AppError err) {
+                    CHECK(err.error_code.value() < 0);
```

Comment:
Are you actually testing an error case here, or is this leftover debugging code?

Reply:
leftover debugging code, I've cleaned it up